src/features/HomePage/index.tsx

```diff
@@ -1,61 +1,238 @@
 import { useNavigate } from 'react-router-dom';
-import { useTranslation } from 'react-i18next';
 import styled from 'styled-components';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 
+import { fetchNotices } from './noticesApi';
+import {
+  Mentor,
+  selectFilteredMentors,
+  useGetMentorsQuery,
+} from '../MentorPage/mentorPageApi';
 import { selectHasUnreadMessages } from '@/features/Chat/chatSlice';
 import { useAppSelector } from '@/store';
+import { useGetLayoutMode } from '@/hooks/useGetLayoutMode';
 
+import Background from '@/static/img/mountain-background.svg';
+import ListCard from '../MentorPage/components/MentorList/MentorCard/List';
+import MentorCard from '../MentorPage/components/MentorList/MentorCard/Expanded';
 import NewMessagesImage from '@/static/img/new-messages.svg';
 import PageWithTransition from '@/components/PageWithTransition';
-import { palette } from '@/components/variables';
+import {
+  NAVIGATION_HEIGHT,
+  OUTER_HORIZONTAL_MARGIN,
+  palette,
+} from '@/components/variables';
 import Text from '@/components/Text';
 import { TextButton } from '@/components/Buttons';
+import { UserRole, selectUserRole } from '../Authentication/userSlice';
+
+import type { Notice } from './noticesApi';
 
 const HomePage = () => {
   const navigate = useNavigate();
   const { t } = useTranslation('home');
 
   const unreadMessagesFound: boolean = useAppSelector(selectHasUnreadMessages);
+  const navigateToAdminPanel = () => navigate('/admin');
   const navigateToChat = () => navigate('/chat');
+  const navigateToMentors = () => navigate('/mentors');
+
+  const userRole: UserRole | null = useAppSelector(selectUserRole);
+
+  // TODO: Mobile view
+  const { isMobile } = useGetLayoutMode();
+  console.log('isMobile is ' + isMobile);
+  const { isLoading } = useGetMentorsQuery();
+  const [selectedMentor, setSelectedMentor] = useState<Mentor | null>(null);
+  const mentorsToShow: Mentor[] = useAppSelector(selectFilteredMentors()).slice(
+    0,
+    2,
+  );
+  const [notices, setNotices] = useState<Notice[]>([]);
+
+  useEffect(() => {
+    const notices: Notice[] = fetchNotices();
+    setNotices(notices);
+  }, [notices]);
```

Comment:
I dont like to run api-requests in effects in our project. We have used the rtk-query pattern in other places too. I would expect us to use the same pattern everywhere, even when mocking data.

Reply:
Very simple api is implemented in src/features/MentorPage/mentorPageApi.ts and it is connected to our state in src/store.ts 

Please use the same pattern. Very nice documentation is also provided in https://redux-toolkit.js.org/tutorials/rtk-query
I'm also happy to help if you have any questions!

Reply:
Yeah absolutely, we'll do this using the same pattern! But the question is, should we do it now or could this be left for after the MVP? 🤔